src/API.jl

```diff
@@ -58,27 +58,29 @@ function develop(ctx::Context, pkgs::Vector{PackageSpec}; shared::Bool=true,
     Context!(ctx; kwargs...)
 
     for pkg in pkgs
-        # if julia is passed as a package the solver gets tricked
-        if pkg.name == "julia"
-            pkgerror("`julia` is not a valid package name.")
+        if pkg.name == "julia" # if julia is passed as a package the solver gets tricked
+            pkgerror("`julia` is not a valid package name")
         end
+        pkg.name === nothing || check_package_name(pkg.name, "develop")
         if pkg.name === nothing && pkg.uuid === nothing && pkg.repo.source === nothing
-            pkgerror("When calling `develop`, packages must be specified by name, UUID, URL, or filesystem path.")
+            pkgerror("name, UUID, URL, or filesystem path specification required when calling `develop`")
         end
         if pkg.repo.rev !== nothing
-            pkgerror("It is invalid to specify a git revision when calling `develop`:",
-                     " `$(pkg.repo.rev)` given for package $(err_rep(pkg)).")
+            pkgerror("git revision specification invalid when calling `develop`:",
+                     " `$(pkg.repo.rev)` specified for package $(err_rep(pkg))")
         end
         if pkg.version != VersionSpec()
-            pkgerror("It is invalid to specify a version when calling `develop`:",
-                     " `$(pkg.version)` given for package $(err_rep(pkg)).")
+            pkgerror("version specification invalid when calling `develop`:",
+                     " `$(pkg.version)` specified for package $(err_rep(pkg))")
         end
     end
 
     new_git = handle_repos_develop!(ctx, pkgs, shared)
 
-    any(pkg -> Types.collides_with_project(ctx, pkg), pkgs) &&
-        pkgerror("Cannot `develop` package with the same name or uuid as the project")
+    for pkg in pkgs
+        Types.collides_with_project(ctx, pkg) || continue
```

Comment:
I would rather just write this as either

```jl
if Types.collides....
    pkgerror()
end
```
I think it is worth the extra line :)

Reply:
:laughing: fair enough